src/ik_parallel.h

```diff
@@ -86,62 +91,83 @@ class ParallelExecutor
     }
 };
 
+std::optional<std::unique_ptr<IKSolver>> makeSolver(const IKParams& params) {
+  if (auto evolution_1 = makeEvolution1Solver(params))
+    return evolution_1;
+  else if (auto evolution_2 = makeEvolution2Solver(params))
+    return evolution_2;
+  else if (auto gradient = makeGradientDecentSolver(params))
+    return gradient;
+  else if (auto test = makeTestSolver(params))
+    return test;
+  else
+    return std::nullopt;
+}
+
 // runs ik on multiple threads until a stop criterion is met
 struct IKParallel
 {
-    IKParams params;
-    std::vector<std::unique_ptr<IKSolver>> solvers;
-    std::vector<std::vector<double>> solver_solutions;
-    std::vector<std::vector<double>> solver_temps;
-    std::vector<int> solver_success;
-    std::vector<double> solver_fitness;
-    int thread_count;
+    IKParams params_;
+    std::vector<std::unique_ptr<IKSolver>> solvers_;
+    std::vector<std::vector<double>> solver_solutions_;
+    std::vector<std::vector<double>> solver_temps_;
+    std::vector<int> solver_success_;
+    std::vector<double> solver_fitness_;
+    size_t thread_count_;
     // std::vector<RobotFK_Fast> fk; // TODO: remove
-    double timeout;
-    bool success;
-    std::atomic<int> finished;
-    std::atomic<uint32_t> iteration_count;
-    std::vector<double> result;
-    std::unique_ptr<ParallelExecutor> par;
-    Problem problem;
-    bool enable_counter;
-    double best_fitness;
+    std::chrono::time_point<std::chrono::system_clock, std::chrono::duration<double>> timeout_;
+    bool success_;
+    std::atomic<int> finished_;
+    std::atomic<uint32_t> iteration_count_;
+    std::vector<double> result_;
+    std::unique_ptr<ParallelExecutor> par_;
+    Problem problem_;
+    bool enable_counter_;
+    double best_fitness_;
 
     IKParallel(const IKParams& params)
-        : params(params)
+        : params_(params)
     {
         // solver class name
-        std::string name = params.solver_class_name;
+        std::string name = params_.solver_class_name;
 
-        enable_counter = params.enable_counter;
+        enable_counter_ = params_.enable_counter;
 
         // create solvers
-        solvers.emplace_back(IKFactory::create(name, params));
-        thread_count = solvers.front()->concurrency();
-        if(params.thread_count) {
-            thread_count = params.thread_count;
+        auto create_solver = [&params]() {
+            auto result = makeSolver(params);
+            if (!result) {
+                throw std::runtime_error("Invalid Solver Name: " +
+                                        params.solver_class_name);
+            }
+            return std::move(*result);
```

Comment:
Why explicitly move here? I believe the compiler is allowed to do an implicit move.

Reply:
Potentially yes. It's worth seeing if it builds without the `std::move`.

Reply:
Leaving alone since `return *result` errors